gobblin-runtime/src/main/java/gobblin/runtime/JobLauncherFactory.java

```diff
@@ -37,18 +38,23 @@
   /**
    * Create a new {@link JobLauncher}.
    *
-   * @param properties Framework configuration properties
-   * @return Newly created {@link JobLauncher}
+   * <p>
+   *   This method will never return a {@code null}.
+   * </p>
+   *
+   * @param properties framework configuration properties
+   * @param jobProps job configuration properties
+   * @return newly created {@link JobLauncher}
    */
-  public static JobLauncher newJobLauncher(Properties properties)
+  public static @Nonnull JobLauncher newJobLauncher(Properties properties, Properties jobProps)
```

Comment:
My understanding is that `properties` refers to system properties and `jobProps` is job level properties? Can we rename `properties` to `systemProps`. This way it is more clear to a reader the difference between the two. I think this may require some refactoring in other classes too.